go/vt/vttablet/tabletserver/vstreamer/planbuilder.go

```diff
@@ -273,6 +275,22 @@ func (plan *Plan) filter(values, result []sqltypes.Value, charsets []collations.
 			if !found {
 				return false, nil
 			}
+		case NotBetween:
+			if filter.Values == nil || len(filter.Values) != 2 {
+				return false, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "expected 2 filter values when performing BETWEEN")
+			}
+			leftFilterValue, rightFilterValue := filter.Values[0], filter.Values[1]
+			isValueLessThanLeftFilter, err := compare(LessThan, values[filter.ColNum], leftFilterValue, plan.env.CollationEnv(), charsets[filter.ColNum])
+			if err != nil {
+				return false, err
```

Comment:
We should wrap this to make it more informative, e.g.:
```suggestion
				return false, vterrors.Wrap(err, "left value within the IN BETWEEN clause is greater than the right value")
```

Reply:
Actually, we don't throw an error if the left value is greater than the right value (this is similar to the MySQL behaviour). Also, here we are comparing the `ColumnValue` with the left-hand filter value. This error branch will only be covered if there's an invalid comparison `Opcode` or `evalengine.NullsafeCompare()` throws an error.